### DISCUSSION TITLE
das2 server vs. Das2Server

### DISCUSSION THREAD

**Discussion by cpiker** [das-developers/discussions#9] [General]
Back in the original 1995 inception, *das* was used as an indefinite article.  It's basically "the" in German.  Since Java classes (and programming tokens in general) can't have spaces, it makes sense to have a class named "DasServer", however in normal English we don't string articles and nouns together into larger words.   So the equivalent program would be a "*das* server" not a "dasserver".

So should a person type "*das2*" versus "*das* generation two"?  Since it saves 14 characters, and provides a hook for branding, I'd vote that "das2", though incorrect English, is reasonable enough.  It's very similar to the precedent set by the usage of "i18n" as a shorthand for the word "internationalization" in the context of porting software interfaces to other languages.  

These things are always judgment calls.  For me, I'm resistant to using programming idioms when writing text for a general audience, and multi-word camel case is a semantic style that is not used in common English writing.  Writing "*das2*", though incorrect, has significant utility, however; "das2server" only saves a single character and that's not enough utility to make it worth breaking the rules.

For common sentences (such as project descriptions) I'd say the string "das2server", and especially "Das2Server", should be considered typos by *das* developers.  

Thoughts?

**Comment by VoyagerPWS** [comment]
Yes, "das2" is essentially a "brand", and "das2 server" specifies the type (brand) of server, while "das2server" would be appropriate for a specific implementation of a das2 server.  (IMHO)
